src/runtime.js

```diff
@@ -105,6 +106,7 @@ export function installRuntimeServices(global) {
  * @param {!Object<string, string>=} opt_initParams
  */
 export function installAmpdocServices(ampdoc, opt_initParams) {
+  installUrlForDoc(ampdoc);
```

Comment:
Do we already do some URL parsing on startup? Wondering if this would impact initial load.

Reply:
Yah, in viewer, doc-info, preconnect, etc. It's all over the place.